### PR TITLE
Add survey list delete dialog

### PR DESCRIPTION
**Pull request by marshmalien**
https://issues.redhat.com/browse/AAP-22463

survey_delete

### CODE REVIEW

**Comment by jerabekjiri** [comment]
Deleting from kebab menu in row actions shows TODO
Screenshot from 2024-04-09 14-26-36

**Review by jerabekjiri — APPROVED** [review]
Other than the missing delete action in the kebab row actions, LGTM :+1: :rocket:

**Review by resoluteCoder — APPROVED** [review]
Pulled and tested locally echo what @nixocio said about kebab. lgtm :tada:

File: frontend/awx/resources/templates/hooks/useDeleteSurveyDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            <PageTable<Spec>
+              key="items"
+              pageItems={pagination.paged}
+              itemCount={questions.length}
+              tableColumns={modalColumns}
+              keyFn={(question) => question.variable}
+              compact
+              errorStateTitle="Error"
+              emptyStateTitle="No items"
+              autoHidePagination={true}
```
[the remark this answers is not in the document]
Comment (marshmalien) [reply frontend/awx/resources/templates/hooks/useDeleteSurveyDialog.tsx:107]: Yes, I'll add translations there.

**Comment by marshmalien** [comment]
@jerabekjiri @resoluteCoder I added a second commit to address the row action kebab delete
@nixocio I updated the dialog styles and marked the title strings for translation 

Screenshot 2024-04-09 at 11 05 21 AM


row_delete

**Comment by resoluteCoder** [comment]
Sweet works great! :+1:

**Review by nixocio — APPROVED** [review]
Good job @marshmalien

**Pull request merged**